_inc/client/plans/single-product-backup.jsx

```diff
@@ -0,0 +1,185 @@
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+import { translate as __ } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { AccentedCard } from './accented-card';
+import Gridicon from 'components/gridicon';
+
+export function SingleProductBackup( props ) {
+	const { sitePlans } = props;
+
+	const [ selectedBackupType, setSelectedBackupType ] = useState( 'real-time' );
+
+	return (
+		<>
+			<h1 className="plans-section__header">Single Products</h1>
+			<h2 className="plans-section__subheader">Just looking for backups? We've got you covered.</h2>
+			<div style={ { display: 'flex', justifyContent: 'center', marginBottom: '10px' } }>
+				<AccentedCard>
+					{ {
+						header: <SingleProductBackupHeader sitePlans={ sitePlans } />,
+						body: (
+							<SingleProductBackupBody
+								selectedBackupType={ selectedBackupType }
+								setSelectedBackupType={ setSelectedBackupType }
+								sitePlans={ sitePlans }
+							/>
+						),
+					} }
+				</AccentedCard>
+			</div>
+		</>
+	);
+}
+
+function SingleProductBackupHeader( props ) {
+	const { sitePlans } = props;
+
+	return (
+		<div
+			style={ {
+				display: 'flex',
+				flexDirection: 'row',
+				justifyContent: 'space-between',
+				alignContent: 'center',
+			} }
+		>
+			<h3>{ __( 'Jetpack Backup' ) }</h3>
+			{ sitePlans && (
+				<PlanPriceDisplay
+					dailyPrice={ sitePlans[ 'daily-backup' ].price.yearly.text }
+					yearlyPrice={ sitePlans[ 'realtime-backup' ].price.yearly.amount }
+				/>
+			) }
+		</div>
+	);
+}
+
+export function PlanPriceDisplay( props ) {
+	const { dailyPrice, yearlyPrice } = props;
+	const perYearPriceRange = `${ dailyPrice }-${ yearlyPrice } /year`;
+
+	return (
+		<div
+			style={ {
+				display: 'flex',
+				flexDirection: 'row',
+				justifyContent: 'space-between',
+				alignContent: 'center',
+			} }
+		>
+			<SlashedPrice />
+			<div className="plans-price__container">
+				<span className="plans-price__span">{ perYearPriceRange }</span>
+			</div>
+		</div>
+	);
+}
+
+function SlashedPrice() {
+	return (
+		<div className="slashed-price__container" style={ { marginRight: '14px' } }>
+			<div className="slashed-price__slash"></div>
+			{ /* TODO: get this from an API or calculate, currently unsure how to get this increased price */ }
+			<div className="slashed-price__price">{ '$15-25' }</div>
+		</div>
+	);
+}
+
+function SingleProductBackupBody( props ) {
+	const { selectedBackupType, setSelectedBackupType, sitePlans } = props;
+
+	const upgradeLinks = {
+		'real-time': 'https://wordpress.com/jetpack/connect/pro',
+		daily: 'https://wordpress.com/jetpack/connect/premium',
+	};
+
+	const upgradeTitles = {
+		'real-time': __( 'Upgrade to Real-Time Backups' ),
+		daily: 'Upgrade to Daily Backups',
+	};
+
+	function handleSelectedBackupTypeChange( event ) {
```

Comment:
This function needs to access `setSelectedBackupType`, which will only exist inside the scope of `SingleProductBackupBody()`. So I cannot think of any way to truly define it only once since it must access `setSelectedBackupType` and it cannot come from the `onChange` event emitted from the browser.

The only other way I can think of would be to switch back to class components here where the function can live on the class instance, but I don't think it's worth the tiny performance gain here.

If you can think of another way let me know, I may be missing something.

Reply:
Yes, I'd suggest making `SingleProductBackupBody` a `React.Component` instead of a functional component. It only makes sense when we have event handlers inside a functional component. It's not about the tiny performance gain, it's about best practices. We don't want to set a bad example - if everyone else starts doing that, it will start affecting performance more.